aya/src/maps/stack_trace.rs

```diff
@@ -174,12 +185,9 @@ impl StackTrace {
     /// You can use [`util::kernel_symbols()`](crate::util::kernel_symbols) to load kernel symbols. For
     /// user-space traces you need to provide the symbols, for example loading
     /// them from debug info.
-    pub fn resolve(&mut self, symbols: &BTreeMap<u64, String>) -> &StackTrace {
+    pub fn resolve<R: SymbolResolver>(&mut self, symbols: &R) -> &StackTrace {
         for frame in self.frames.iter_mut() {
-            frame.symbol_name = symbols
-                .range(..=frame.ip)
-                .next_back()
-                .map(|(_, s)| s.clone())
+            frame.symbol_name = symbols.resolve_symbol(frame.ip).map(|s| s.to_string())
```

Comment:
s/to_string/to_owned/

Reply:
I don't think I understand; `str::to_owned` returns `String`. https://doc.rust-lang.org/stable/src/alloc/str.rs.html#205

Reply:
```
[addisoncrump@fuzzbox aya]$ cargo build
   Compiling aya v0.11.0 (/home/addisoncrump/git/aya/aya)
error[E0308]: mismatched types
   --> aya/src/maps/stack_trace.rs:190:33
    |
190 |             frame.symbol_name = symbols.resolve_symbol(frame.ip).map(|s| s.to_owned())
    |             -----------------   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `Option<String>`, found `Option<Cow<'_, str>>`
    |             |
    |             expected due to the type of this binding
    |
    = note: expected enum `Option<String>`
               found enum `Option<Cow<'_, str>>`
```

Reply:
Ah, it's [`into_owned`](https://doc.rust-lang.org/stable/std/borrow/enum.Cow.html#method.into_owned), not `to_owned`.

Reply:
Resolved